### PR TITLE
Print a helpful message for JSON column update failures.

### PR DESCRIPTION
**Pull request by agnivade**
Refer: https://github.com/mattermost/mattermost-server/issues/18038

```release-note
NONE
```

### CODE REVIEW

File: store/sqlstore/store.go [hunk 1]
```diff
@@ -920,7 +920,11 @@ func (ss *SqlStore) AlterColumnTypeIfExists(tableName string, columnName string,
 	}
 
 	if err != nil {
-		mlog.Critical("Failed to alter column type", mlog.Err(err))
+		msg := "Failed to alter column type."
+		if mySqlColType == "JSON" && postgresColType == "jsonb" {
+			msg += " It is likely you might have invalid JSON values in the column. Please fix the values manually and run the migration again."
```
Comment (agnivade) [inline store/sqlstore/store.go:925]: 0/5 on a better error message here. Let me know if you want something changed.

Comment (streamer45) [reply store/sqlstore/store.go:925]: I'd remove the "might" since it's already a conditional but other than that sounds good to me.

**Review by streamer45 — APPROVED** [review]
Thanks, this will probably avoid a few community issues down the line.

File: store/sqlstore/store.go (hunk 1 above)
Comment (agnivade) [reply store/sqlstore/store.go:925]:
```suggestion
			msg += " It is likely you have invalid JSON values in the column. Please fix the values manually and run the migration again."
```

**Pull request merged**